### PR TITLE
Remove `toolchain --install/--uninstall/--install-dir`

### LINKED ISSUE #1612: alr toolchain --install / --install-dir now obsolete
Hi @mosteo, as part of V2 breaking changes I think we mentioned removing the install part of the toolchain command in favor of the new `alr install` command.

Is there any reason to keep that in?

### PR DESCRIPTION
**Pull request by mosteo**
Fixes #1612

### CODE REVIEW

File: src/alire/alire-toolchains.adb [hunk 1]
```diff
@@ -487,6 +487,17 @@ package body Alire.Toolchains is
...
+      --  Remove caching of external condition too for consistency
+
+      if CLIC.Config.Defined (Config.DB.all, Tool_Key (Crate, For_Is_External))
+      then
+         Trace.Debug
+           ("Unsetting " & Tool_Key (Crate, For_Is_External) & ": "
+            & CLIC.Config.Edit.Unset
+              (Config.Edit.Filepath (Level),
+               Tool_Key (Crate, For_Is_External))'Image);
+      end if;
```
Comment (mosteo) [inline src/alire/alire-toolchains.adb:500]: This is a bugfix for a corner case when the user selected None in the assistant after having selected a regular compiler, causing detection of the external compiler to be skipped. It was exposed when tweaking a test to remove `--uninstall`.

File: testsuite/tests/toolchain/custom_install_dir/test.py [hunk 2]
```diff
@@ -1,44 +1,53 @@
 """
 Check folders when installing binary compiler crates at a custom location
+through `alr get`
```
Comment (Fabien-Chouteau) [inline testsuite/tests/toolchain/custom_install_dir/test.py:3]: Why not `alr install --prefix` here?

File: doc/user-changes.md [hunk 3]
```diff
@@ -6,6 +6,19 @@ stay on top of `alr` new features.
 
 ## Release `2.0-dev`
 
+### Deprecation of `toolchain --install/--uninstall/--install-dir`
+
+PR [#1614](https://github.com/alire-project/alire/pull/1614)
+
+Toolchain installation for use by Alire is still possible using
+`alr toolchain --select`. For the installation of toolchains outside of Alire
+management, `alr get` covers the `alr toolchain --install --install-dir` use
```
Comment (Fabien-Chouteau) [inline doc/user-changes.md]: Same as below, why not recommend `alr install --prefix` here?

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]
So I worked on the same thing in parallel :man_facepalming: 

In my commit I renamed `alr toolchain` to `alr default-toolchain` because I think it will remove some confusion on how Alire handles toolchains.

**Comment by mosteo** [comment]
> So I worked on the same thing in parallel 🤦‍♂️

:-(
 
> In my commit I renamed `alr toolchain` to `alr default-toolchain` because I think it will remove some confusion on how Alire handles toolchains.

This will break a lot of workflows/actions out there, ours included... Do you think there's that much risk of confusion, given the `--select`?

File: testsuite/tests/toolchain/custom_install_dir/test.py (hunk 2 above)
Comment (mosteo) [reply testsuite/tests/toolchain/custom_install_dir/test.py:3]: Just because it is the closest to what the test was doing. We could add a new test that checks a similar idea but with `alr install`.

File: doc/user-changes.md (hunk 3 above)
Comment (mosteo) [reply doc/user-changes.md]: Again, because for people that was using `--install-dir`, this is what they would find more similar. But if we want people to just use `alr install`, we can omit `alr get` altogether.

File: testsuite/tests/toolchain/custom_install_dir/test.py (hunk 2 above)
Comment (Fabien-Chouteau) [reply testsuite/tests/toolchain/custom_install_dir/test.py:3]: `alr toolchain --install` was  introduced for people that wanted to get a toolchain from the Alire index but not use it in the Alire context. 

I feel like `alr install` is a better option for this use case.

**Comment by Fabien-Chouteau** [comment]
> > In my commit I renamed `alr toolchain` to `alr default-toolchain` because I think it will remove some confusion on how Alire handles toolchains.
> 
> This will break a lot of workflows/actions out there, ours included... 

That's why we are doing a major release ^^

> Do you think there's that much risk of confusion, given the `--select`?

I think we do see misconceptions from time to time.

Which makes me thing we could also take this opportunity to rename `alr config` into `alr settings` (see #1607 ).

File: testsuite/tests/toolchain/custom_install_dir/test.py (hunk 2 above)
Comment (mosteo) [reply testsuite/tests/toolchain/custom_install_dir/test.py:3]: Sure, I'll adapt the test.

**Comment by mosteo** [comment]
> > > In my commit I renamed `alr toolchain` to `alr default-toolchain` because I think it will remove some confusion on how Alire handles toolchains.
> > 
> > 
> > This will break a lot of workflows/actions out there, ours included...
> 
> That's why we are doing a major release ^^

I feel this is a point where I must resist, and not so much for the brokenness but because I find `default-toolchain` unnecessarily long, a bit ugly, and itself a bit misleading as for example when several are installed, the output shows both default and non-default toolchains:
```
$ alr toolchain
CRATE         VERSION       STATUS    NOTES                                     
gnat_native   11.2.4        Available                                           
gnat_native   12.2.1        Available                                           
gnat_native   13.2.1        Default
``` 

I'd be less objecting to `alr toolchain --select-default` or even `alr toolchain --set-default`.

But I have no more arguments so if you're truly convinced this will be a net gain, I won't object further.

> Which makes me thing we could also take this opportunity to rename `alr config` into `alr settings` (see #1607 ).

Sounds good, I saw the exchange with Maxim.

I'm a bit uneasy about introducing such breaking changes so late in the RC period, in particular if we dont have a 2nd RC (which I didn't plan to have). What do you think of having the old name as an undocumented alias that emits a warning pointing to the new command?

**Comment by mosteo** [comment]
The last push completes the removal. Any command renamings I would do in a separate PR.

**Review by Fabien-Chouteau — APPROVED** [review]

**Comment by Fabien-Chouteau** [comment]
@mosteo I started working on the `alr settings` modification.
